include/boost/geometry/algorithms/detail/buffer/buffered_piece_collection.hpp

```diff
@@ -562,9 +561,9 @@ struct buffered_piece_collection
                 }
                 // Walk through them, in reverse to insert at right index
                 int index_offset = pc.robust_turns.size() - 1;
-                for (typename std::vector<robust_turn>::const_reverse_iterator
-                        rit = pc.robust_turns.rbegin();
-                    rit != pc.robust_turns.rend();
+                for (typename boost::range_reverse_iterator<const std::vector<robust_turn> >::type
+                        rit = boost::const_rbegin(pc.robust_turns);
+                    rit != boost::const_rend(pc.robust_turns);
                     ++rit, --index_offset)
                 {
                     int const index_in_vector = 1 + rit->seg_id.segment_index - piece_segment_index;
```

Comment:
It fails in the `rit != pc.robust_turns.rend()` line. AFAIU it's because the compiler doesn't explicitly convert the mutable reverse iterator to const reverse iterator before comparison. `rbegin()` and `rend()` return mutable reverse iterators when the container is mutable, which is this case. Those mutable iterators must then be converted into const iterators in the original code. So in the above case the implicit conversions are ommited, const reverse iterators are created explicitly, which should help.